OSDOCS-4438: Deploy Microshift with mirror registry (#64852)
openshift/openshift-docs: modules/microshift-configuring-hosts-for-mirror.adoc

// Module included in the following assemblies:
//
// * microshift/running_applications/microshift-deploy-with-mirror-registry.adoc

:_content-type: PROCEDURE
[id="microshift-configuring-hosts-for-mirror_{context}"]
= Configuring hosts for mirror registry access

To configure a {microshift-short} host to use a mirror registry, you must give the {microshift-short} host access to the registry by creating a configuration file that maps the Red Hat registry host names to the mirror.

.Prerequisites
* Your mirror host has access to the internet.
* The mirror host can access the mirror registry.
* You configured the mirror registry for use in your restricted network.
* You downloaded the pull secret and modified it to include authentication to your mirror repository.

.Procedure
. Log into your {microshift-short} host.

. Enable the SSL certificate trust on any host accessing the mirror registry by completing the following steps:

> .. Copy the `rootCA.pem` file from the mirror registry to the MicroShift host at the `/etc/pki/ca-trust/source/anchors` directory.

Reviewer: Could this step be a little more explicit about the location of rootCA.pem. I found it on <registry_path>/quay-rootCA.

// Module included in the following assemblies:
//
// * microshift/running_applications/microshift-deploy-with-mirror-registry.adoc

:_content-type: PROCEDURE
[id="microshift-configuring-hosts-for-mirror_{context}"]
= Configuring hosts for mirror registry access

> To configure a {product-title} host to use a mirror registry, you must give the host access and create a configuration file with the host name and port through which the registry can be reached.

Reviewer 2: ```suggestion
To configure a {product-title} host to use a mirror registry, you must give the host access to the registry by creating a configuration file mapping the Red Hat registry host names to the mirror.
```

.Prerequisites

> * Your mirror host has access to the internet.

Reviewer 2: Mirror is air-gapped most of the times.

Reviewer 2: Did you mean here that MicroShift host should have network access to the mirror?

Author: I was looking at, "Download Images on a host with the Internet access." https://docs.google.com/document/d/1aqgjxP2YqIGi2djU96vN2yykDZpIC8s2hg4QHTCz390/edit#heading=h.pawolv522s21

Reviewer 2: OK, I was confused by names. To reiterate, we have 3 entities involved
- mirror host - with Internet access
- mirror registry - no Internet access
- microshift host - access to mirror registry
Looks like the above prerequisite is OK.

> * You configured a mirror registry to use in your restricted network and can access the certificate and credentials that you configured.

Reviewer 2: I'm not sure what that part after "and" means.

Author: Whoo, that was end-of-day sentence. I think I meant,

* Your mirror host has access to the internet.
* The mirror host can access the mirror registry.
* You configured a mirror registry to use in your restricted network.
* You downloaded the pull secret and modified it to include authentication to your mirror repository.

I must have been thinking of the authentication needed to pull the images.

Reviewer 2: Yes, let's replace the original text with the 4 bullet points you mentioned in the comment. I think it's great.
